ldapauthenticator/ldapauthenticator.py

```diff
@@ -529,8 +538,30 @@ async def authenticate(self, handler, data):
         if not self.use_lookup_dn_username:
             username = data["username"]
 
-        user_info = self.get_user_attributes(conn, userdn)
-        if user_info:
-            self.log.debug("username:%s attributes:%s", username, user_info)
-            return {"name": username, "auth_state": user_info}
-        return username
+        user_attrs = self.get_user_attributes(conn, userdn)
+        auth_state = {
+            "ldap_groups": ldap_groups,
+            "user_attrs": user_attrs,
+        }
+        self.log.debug("username:%s attributes:%s", username, user_attrs)
+        return {"name": username, "auth_state": auth_state}
+
+    async def check_allowed(self, username, auth_model):
+        allowed = super().check_allowed(username, auth_model)
+        if isawaitable(allowed):
+            allowed = await allowed
+        if allowed is True:
+            return True
+        if self.search_filter and not self.allowed_groups:
+            # search_filter was specified
+            # consider matching this filter sufficient to allow access
+            # _IF_ allowed_groups is unspecified
+            return True
```

Comment:
`search_filter` isn't applied at all times, only if `lookup_dn` is used, so if someone misconfigures `search_filter` with `lookup_dn=False`, this can lead to a user being allowed via `search_filter` without matching the filter.